### ISSUE TITLE
Not working if code contains Optional Chaining Operator

### ISSUE THREAD

**Issue by sankarkumar23** [Josee9988/MinifyAll#65]
Looks like javascript's Optional Chaining Operator "?." has not been incorporated.
Giving out error if code contains "?."
Appreciate if this can be fixed ASAP.

**Comment by Josee9988** [comment]
TL;DR

The JS minimization is done by an external package called [terser](https://github.com/terser/terser) which is out of my hands and it is not developed by me.


---


Hi, JavaScript minimization is done witj a 3rd party package, which it is not developed by me.
The rest of the languages are fully made by me but not the JS one.

The terser's JS minimization is an extra feature for the extension MinifyAll, but not a part of the [MinifyAll core](https://github.com/Josee9988/MinifyAllCli).

This issue should be posted in the [terser's issue page](https://github.com/terser/terser/issues).


Thanks for contributing

**Comment by sankarkumar23** [comment]
Thanks for the information.

**Comment by Josee9988** [comment]
Anyways, I will update the terser package to obtain the latest features.

**Comment by sankarkumar23** [comment]
After contacting terser this feature has been incorporated with v5.2.0.
https://github.com/Josee9988/MinifyAll/blob/8a350fb1c45b981e1f41cf8bc3b0fb71dc3e2927/package.json#L263
Can I update to latest terser version in my local machine.. until you update your package. If yes, please explain how?

**Comment by Josee9988** [comment]
I will try to publish a new version tomorrow.

If you need it now, go to the VSCode extension folder, search the MinifyAll folder, then open a terminal and write:

`
npm update terser
`

**Comment by sankarkumar23** [comment]
Awesome.. If it can happen very soon.. I'll wait. Thank a lot for all your support.

**Comment by Josee9988** [comment]
I'll comment again when the terser package is updatedm

**Comment by Josee9988** [comment]
Update the MinifyAll extension to its latest version (2.4.7).

Terser was updated to 5.3.2.

**Comment by sankarkumar23** [comment]
> Update the MinifyAll extension to its latest version (2.4.7).
> 
> Terser was updated to 5.3.2.

Not sure If I am missing something.. after upgrade minify is completely failing.. it is generating a min file with no content.. just showing undefined.

**Comment by Josee9988** [comment]
Please post a new issue with detailed information/media prove about this new issue.
